### PR TITLE
feat: support autoconfigure WeightedServiceInstanceListSupplier

### PR DESCRIPTION
**Pull request by jizhuozhi**
Enable configuring `WeightedServiceInstanceListSupplier` via `spring.cloud.loadbalancer.configurations`

### CODE REVIEW

**Comment by jizhuozhi** [comment]
Hello, @OlgaMaciaszek , PLAT, thanks.

File: docs/src/main/asciidoc/spring-cloud-commons.adoc [hunk 1]
```diff
@@ -935,7 +935,7 @@ By default, we try to read and parse the weight from the metadata map (the key i
 
 If the weight is not specified in the metadata map, we default the weight of this instance to be 1.
 
-You can use this sample configuration to set it up:
+You can configure it either by setting the value of `spring.cloud.loadbalancer.configurations` to `weighted` or by providing your own `ServiceInstanceListSupplier` bean -- for example:
```
Comment (OlgaMaciaszek) [inline docs/src/main/asciidoc/spring-cloud-commons.adoc:938]: Please reword "bean -- for example:" to "bean, for example:".

**Review by OlgaMaciaszek — CHANGES_REQUESTED** [review]
@jizhuozhi Thanks for submitting this PR. There's just a minor thing in the doc to fix. Have added a comment - please address it.

**Comment by jizhuozhi** [comment]
Hello, @OlgaMaciaszek . Document has fixed, but it's copied from other chapters, we should also fix them :).

**Pull request merged**